src/Controller/ArticlesController.php

```diff
@@ -910,4 +928,19 @@ private function generateFiguresPath(ArticleHistory $history, int $forVersion =
 
         return $this->get('router')->generate('article-version-figures', [$currentVersion, 'version' => $forVersion]);
     }
+
+    private function generateXmlPath(ArticleHistory $history, int $forVersion = null, string $fragment = null) : string
```

Comment:
`fragment` seems to be always `null`, and it shouldn't be used on an XML file right? (If it's the URL fragment `#something`)

Reply:
Should look at removing the duplication with `generateTextPath` and `generateFiguresPath`.